invenio_accounts/datastore.py

```diff
@@ -19,14 +23,46 @@
 class SessionAwareSQLAlchemyUserDatastore(SQLAlchemyUserDatastore):
     """Datastore which deletes active session when a user is deactivated."""
 
+    def verify_user(self, user):
+        """Verify a user."""
+        now = datetime.utcnow()
+        user.blocked_at = None
+        user.verified_at = now
+        if user.confirmed_at is None:
+            user.confirmed_at = now
+        if not user.active:
+            user.active = True
```

Comment:
```suggestion
        user.confirmed_at = user.confirmed_at or now
        user.active = True # Directly set to True, assuming verification implies activation
```

Reply:
Your concern is valid If there's a chance that `user.confirmed_at ` could be mistakenly set to a falsy value after it's been correctly set to a timestamp once, then the `or` expression would indeed be dangerous because it would then reset `user.confirmed_at ` to `now`.
 But with normal use and validation applies to this db column, assuming `user.confirmed_at `  is either None (not set) or a valid timestamp (meaning it has a truthy value), the `or` expression to me is safe and does exactly what's intended unless there are other scenarios I am not aware of!

Reply:
Updated and also the block _user is updated with the same issue.